network/ip_tracker.go

```diff
@@ -461,10 +587,56 @@ func (i *ipTracker) resetBloom() error {
 	i.bloomSalt = newSalt
 	i.maxBloomCount = bloom.EstimateCount(numHashes, numEntries, maxFalsePositiveProbability)
 
-	for nodeID, ip := range i.mostRecentTrackedIPs {
-		bloom.Add(newFilter, ip.GossipID[:], newSalt)
+	for nodeID, trackedNode := range i.tracked {
+		if trackedNode.ip == nil {
+			continue
+		}
+
+		bloom.Add(newFilter, trackedNode.ip.GossipID[:], newSalt)
 		i.bloomAdditions[nodeID] = 1
 	}
 	i.bloomMetrics.Reset(newFilter, i.maxBloomCount)
 	return nil
 }
+
+func getGossipableIPs[T any](
+	i *ipTracker,
+	iter map[ids.ID]T, // The values in this map aren't actually used.
+	allowed func(ids.ID) bool,
+	exceptNodeID ids.NodeID,
+	exceptIPs *bloom.ReadFilter,
+	salt []byte,
+	maxNumIPs int,
+) []*ips.ClaimedIPPort {
```

Comment:
We should limit how many IPs we are willing to check in a single request. But I don't think that needs to be in this PR.